packages-ts/gauntlet-terra-contracts/src/commands/abstract/batchWrapper.ts

```diff
@@ -0,0 +1,62 @@
+import AbstractCommand, { makeAbstractCommand } from '.'
+import { Result, WriteCommand } from '@chainlink/gauntlet-core'
+import { TerraCommand, TransactionResponse, logger } from '@chainlink/gauntlet-terra'
+import { AccAddress, LCDClient, Msg } from '@terra-money/terra.js'
+import { prompt } from '@chainlink/gauntlet-core/dist/utils'
+import { ExecutionContext } from './executionWrapper'
+import { parseJSON } from '@chainlink/gauntlet-terra/dist/lib/rdd'
+
+export const wrapCommand = (command) => {
+  return class BatchCommand extends TerraCommand {
+    static id = `${command.id}:batch`
+    subCommands: any[]
+
+    constructor(flags, args) {
+      super(flags, args)
+    }
+
+    buildCommand = async (flags, args): Promise<TerraCommand> => {
+      const input = flags.input ? flags.input : parseJSON(flags.inputFile, 'BatchInputFile')
+
+      this.subCommands = await Promise.all(
+        input.map(async (individualInput) => {
+          const newFlags = { ...flags, input: individualInput }
+
+          let c = new command(newFlags, args) as TerraCommand
+          await c.invokeMiddlewares(c, c.middlewares)
+          c = c.buildCommand ? await c.buildCommand(newFlags, args) : c
```

Comment:
I'm not sure if we should combine both features like that. I think it might be cleaner to have two `batchWrapper` classes, where one is `<action>:batch:single` and the other is `<action>:batch:multiple`

Reply:
Let's just do that with a flag, no need for another command type. Including the `--multiple` flag could link the input to its arg

Reply:
kk sounds good to me